libclc/generic/libspirv/float16.cl

```diff
@@ -4735,9 +4735,19 @@ __spirv_ocl_fract(__clc_float16_t args_0, __clc_float16_t __global *args_1) {
   return __spirv_ocl_fract(as_half(args_0), (__clc_fp16_t __global *)(args_1));
 }
 
-_CLC_OVERLOAD _CLC_DEF __clc_vec2_fp16_t
-__spirv_ocl_fract(__clc_vec2_float16_t args_0, __clc_vec2_float16_t *args_1) {
-  return __spirv_ocl_fract(as_half2(args_0), (__clc_vec2_fp16_t *)(args_1));
+#if __OPENCL_C_VERSION__ == CL_VERSION_2_0 ||                                  \
+    (__OPENCL_C_VERSION__ >= CL_VERSION_3_0 &&                                 \
+     defined(__opencl_c_generic_address_space))
```

Comment:
As is, I think it is fairly readable, but personally I think it would be as readable (and shorter) if we defined another macro, for example `_CLC_GENERIC_AS_SUPPORTED`, based on the above requirements and use that instead. Just a small suggestion, so if you prefer this I won't put up a fight.

Reply:
Yeah, I agree. Though we should probably try and place it centrally. Maybe in a central header?

Since it'd be best to refactor more files at the same time, I could perhaps do that in a follow-up commit?

Reply:
Sounds good to me! 😄

Reply:
I'm on it!